src/NovelAudioService.cpp

```diff
@@ -0,0 +1,173 @@
+//
+// Created by krjoh on 29/08/2019.
+//
+
+#include "NovelAudioService.h"
+#include <iostream>
+#include <functional>
+
+namespace NovelRT {
+
+NovelAudioService::NovelAudioService(){
+  initializeAudio();
+  _nextChannel = 0;
+}
+
+int NovelAudioService::initializeAudio(){
+  if (SDL_Init(SDL_INIT_AUDIO) < 0 )
+  {
+    std::cout << "ERROR: Cannot play audio!" << std::endl;
+    std::cout << "SDL Error: " << SDL_GetError() << std::endl;
+    isInitialized = false;
+    return -1;
+  }
+  if (Mix_OpenAudio(44100, MIX_DEFAULT_FORMAT, 2, 2048) < 0)
+  {
+    std::cout << "ERROR: Cannot play audio!" << std::endl;
+    std::cout << "SDL_Mixer Error: " << Mix_GetError() << std::endl;
+    isInitialized = false;
+    return -1;
+  }
+  std::cout << "INFO: Allocating channels..." << std::endl;
+  if (Mix_AllocateChannels(2) < 0)
+  {
+    std::cout << "ERROR: Failed to allocate channels." << std::endl;
+    isInitialized = false;
+    return -1;
+  }
+  std::cout << "INFO: Mixer engaged!" << std::endl;
+
+  return 1;
+}
+
+void NovelAudioService::loadSound(std::string input) {
+  auto exists = _sounds.find(input);
+  if (exists != _sounds.end()) return;
+
+  Mix_Chunk* newSound = Mix_LoadWAV(input.c_str());
+  if (newSound != nullptr)
+  {
+    _sounds[input] = newSound;
+  }
+
+}
+
+void NovelAudioService::unloadSound(std::string input) {
+  auto existingSound = _sounds.find(input);
+  if (existingSound == _sounds.end()) return;
+
+  Mix_FreeChunk(_sounds[input]);
+  _sounds.erase(existingSound);
+}
+
+void NovelAudioService::loadMusic(std::string input) {
+  auto exists = _music.find(input);
+  if (exists != _music.end()) return;
+
+  Mix_Music* newMusic = Mix_LoadMUS(input.c_str());
+  if (newMusic != nullptr)
+  {
+    _music[input] = newMusic;
+  }
+}
+
+void NovelAudioService::unloadMusic(std::string input) {
+  auto existingMusic = _music.find(input);
+  if (existingMusic == _music.end()) return;
+
+  Mix_FreeMusic(_music[input]);
+  _music.erase(existingMusic);
+}
+
+//Specifying the filename for playMusic() infers that it has not been played
+//or that the music should be switched.
+void NovelAudioService::playMusic(std::string musicName) {
+  auto existingMusic = _music.find(musicName);
+  if (existingMusic == _music.end())
+  {
+    loadMusic(musicName);
+    existingMusic = _music.find(musicName);
+    if (existingMusic == _music.end())
+    {
+      return;
+    }
+  }
+
+  Mix_PlayMusic(_music[musicName],-1);
+}
+
+//Not specifying a filename assumes that the music channel has been paused.
+//We doublecheck to make sure that is the case
+void NovelAudioService::playMusic() {
+  if (Mix_PausedMusic() == 1)
```

Comment:
maybe invert this?

Reply:
This should not be inverted to resume, as it will always return 1 or 0, 1 being that the Music channel is paused. One-lined it though and removed bracing.